### ISSUE TITLE
pip install is broken

### ISSUE THREAD

**Issue by mpagot** [openSUSE/osc#1270]
**Describe the bug**
```
% pip install osc

...

      creating build/scripts-3.10
      copying and adjusting osc-wrapper.py -> build/scripts-3.10
      changing mode of build/scripts-3.10/osc-wrapper.py from 644 to 755
      generating build/osc.1.gz
      Traceback (most recent call last):
        File "/tmp/pip-install-u5vh3lxa/osc_257ef0578fb142abbd807a261d0f2226/osc/commandline.py", line 9273, in _load_plugins
          mod = imp.load_source(modname, os.path.join(plugin_dir, extfile))
        File "/usr/lib64/python3.10/imp.py", line 172, in load_source
          module = _load(spec)
        File "<frozen importlib._bootstrap>", line 719, in _load
        File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 883, in exec_module
        File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
        File "/usr/lib/osc-plugins/qam.py", line 9, in <module>
          from oscqam import strict_version
      ModuleNotFoundError: No module named 'oscqam'

      /usr/lib/osc-plugins/qam.py: No module named 'oscqam'

       Try 'env OSC_PLUGIN_FAIL_IGNORE=1 osc ...'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> osc

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Versions**
- Operating system: `openSUSE Tumbleweed`
- Packages: `osc-1.0.0~b3-1.1.noarch`  but it is not relevan as bug is about a different installation method

**To Reproduce**
1. Run command `pip install osc==0.167.2` and `pip install osc==0.182.0` and `pip install osc`
2. See error
```
Collecting osc==0.167.2
  Downloading osc-0.167.2.tar.gz (261 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 262.0/262.0 kB 2.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Requirement already satisfied: M2Crypto in ./.venv/lib/python3.10/site-packages (from osc==0.167.2) (0.38.0)
Collecting chardet
  Downloading chardet-5.1.0-py3-none-any.whl (199 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 199.1/199.1 kB 5.0 MB/s eta 0:00:00
Installing collected packages: chardet, osc
  DEPRECATION: osc is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' opti
on. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for osc ... error
  error: subprocess-exited-with-error

  × Running setup.py install for osc did not run successfully.
  │ exit code: 1
  ╰─> [56 lines of output]
      running install
      /home/michelepa/p_repo/openqa/qem-bot/.venv/lib64/python3.10/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib
      creating build/lib/osc
      copying osc/oscsslexcp.py -> build/lib/osc
      copying osc/oscssl.py -> build/lib/osc
      copying osc/oscerr.py -> build/lib/osc
      copying osc/meter.py -> build/lib/osc
      copying osc/grabber.py -> build/lib/osc
      copying osc/fetch.py -> build/lib/osc
      copying osc/credentials.py -> build/lib/osc
      copying osc/core.py -> build/lib/osc
      copying osc/conf.py -> build/lib/osc
      copying osc/commandline.py -> build/lib/osc
      copying osc/cmdln.py -> build/lib/osc
      copying osc/checker.py -> build/lib/osc
      copying osc/build.py -> build/lib/osc
      copying osc/babysitter.py -> build/lib/osc
      copying osc/__init__.py -> build/lib/osc
      copying osc/OscConfigParser.py -> build/lib/osc
      creating build/lib/osc/util
      copying osc/util/safewriter.py -> build/lib/osc/util
      copying osc/util/rpmquery.py -> build/lib/osc/util
      copying osc/util/repodata.py -> build/lib/osc/util
      copying osc/util/packagequery.py -> build/lib/osc/util
      copying osc/util/helper.py -> build/lib/osc/util
      copying osc/util/debquery.py -> build/lib/osc/util
      copying osc/util/cpio.py -> build/lib/osc/util
      copying osc/util/archquery.py -> build/lib/osc/util
      copying osc/util/ar.py -> build/lib/osc/util
      copying osc/util/__init__.py -> build/lib/osc/util
      running build_scripts
      creating build/scripts-3.10
      copying and adjusting osc-wrapper.py -> build/scripts-3.10
      changing mode of build/scripts-3.10/osc-wrapper.py from 644 to 755
      generating build/osc.1.gz
      Traceback (most recent call last):
        File "/tmp/pip-install-u5vh3lxa/osc_257ef0578fb142abbd807a261d0f2226/osc/commandline.py", line 9273, in _load_plugins
          mod = imp.load_source(modname, os.path.join(plugin_dir, extfile))
        File "/usr/lib64/python3.10/imp.py", line 172, in load_source
          module = _load(spec)
        File "<frozen importlib._bootstrap>", line 719, in _load
        File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 883, in exec_module
        File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
        File "/usr/lib/osc-plugins/qam.py", line 9, in <module>
          from oscqam import strict_version
      ModuleNotFoundError: No module named 'oscqam'

      /usr/lib/osc-plugins/qam.py: No module named 'oscqam'

       Try 'env OSC_PLUGIN_FAIL_IGNORE=1 osc ...'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> osc

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Expected behavior**
Install successfully or pip report that `osc` does not exist

**Additional context**
Even if this project does not suggest to use `pip` for the installation, the package is available in pipy https://pypi.org/project/osc/  and so there are projects that expect this installation method to work like https://github.com/openSUSE/qem-bot/blob/563e0ef8e8a1397bdb9a2a8bd8616cc164e20719/requirements.txt#L1

**Comment by dmach** [comment]
This seems to be caused by 'qam' osc plugin you have on your system.
Can you check if the plugin is properly installed?
Is it updated to the latest version?
How was it installed? Using pip? From a rpm package?

**Comment by dmach** [comment]
@mpagot could you try again with osc-1.0.0 that is on pypi for about a week?

**Comment by mpagot** [comment]
Seems fine to me now

```
% cd /tmp


% python3.10 -m venv .venv


% source .venv/bin/activate


(.venv) % python --version

Python 3.10.10


(.venv)  % pip --version

pip 22.3.1 from /tmp/.venv/lib64/python3.10/site-packages/pip (python 3.10)


(.venv) % pip install osc

Collecting osc
  Downloading osc-1.0.0.tar.gz (248 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 248.0/248.0 kB 2.4 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Collecting cryptography
  Using cached cryptography-39.0.2-cp36-abi3-manylinux_2_28_x86_64.whl (4.2 MB)
Collecting rpm
  Downloading rpm-0.1.0-py3-none-any.whl (4.0 kB)
Collecting urllib3
  Using cached urllib3-1.26.15-py2.py3-none-any.whl (140 kB)
Collecting cffi>=1.12
  Using cached cffi-1.15.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (441 kB)
Collecting pycparser
  Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
Installing collected packages: urllib3, rpm, pycparser, cffi, cryptography, osc
  DEPRECATION: osc is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for osc ... done
Successfully installed cffi-1.15.1 cryptography-39.0.2 osc-1.0.0 pycparser-2.21 rpm-0.1.0 urllib3-1.26.15

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: pip install --upgrade pip



(.venv) % pip freeze

cffi==1.15.1
cryptography==39.0.2
osc==1.0.0
pycparser==2.21
rpm==0.1.0
urllib3==1.26.15
```
